From badf2dbc86ea694d64d593b31cb4e5d4ccb5bc2d Mon Sep 17 00:00:00 2001
From: Massimo Cora <mcora@src.gnome...>
Date: Sat, 21 Mar 2009 11:54:34 +0000
Subject: fixed #576057 – doesn't build with graphviz 2.22.2 (backport from trunk)

	* plugins/class-inheritance/class-inherit.c
	(cls_inherit_draw_graph):
	* plugins/profiler/gprof-function-call-chart-view.c
	(async_draw_graph):
	fixed #576057 – doesn't build with graphviz 2.22.2 (backport from trunk)

svn path=/branches/gnome-2-26/; revision=4886

Backported to 2.24 for node => graph_node renaming change in 2.26
---
diff --git a/plugins/class-inheritance/class-inherit.c b/plugins/class-inheritance/class-inherit.c
index d7346f1..88a3004 100644
--- a/plugins/class-inheritance/class-inherit.c
+++ b/plugins/class-inheritance/class-inherit.c
@@ -815,9 +815,18 @@ cls_inherit_draw_graph (AnjutaClassInheritance *plugin)
 		gdouble node_width;
 		gdouble node_height;
 		point node_pos;
+#ifndef ND_coord_i
+		pointf node_posf;
+#endif
 		
 		/* get some infos from the node */
+#ifdef ND_coord_i
 		node_pos = ND_coord_i(node);
+#else
+		node_posf = ND_coord(node);
+		PF2P(node_posf,node_pos);
+#endif
+		
 		node_width = ND_width (node);
 		node_height = ND_height (node);
 
diff --git a/plugins/profiler/gprof-function-call-chart-view.c b/plugins/profiler/gprof-function-call-chart-view.c
index 315cee2..3070d46 100644
--- a/plugins/profiler/gprof-function-call-chart-view.c
+++ b/plugins/profiler/gprof-function-call-chart-view.c
@@ -530,6 +530,9 @@ async_draw_graph (gpointer user_data)
 	Agedge_t *current_edge;
 	Agraph_t *current_graph;
 	point node_pos;
+#ifndef ND_coord_i	
+	pointf node_posf;
+#endif	
 	gdouble node_width;
 	gdouble node_height;
 	
@@ -541,7 +544,13 @@ async_draw_graph (gpointer user_data)
 		
 		if (self->priv->current_node)
 		{
+#ifdef ND_coord_i			
 			node_pos = ND_coord_i (self->priv->current_node);
+#else
+			node_posf = ND_coord (self->priv->current_node);
+			PF2P(node_posf,node_pos);
+#endif			
+			
 			node_pos.y += self->priv->y_offset;
 			node_width = ND_width (self->priv->current_node);
 			node_height = ND_height (self->priv->current_node);
--
cgit v0.8.2
